pycontrails/core/vector.py

```diff
@@ -1182,14 +1265,19 @@ def __init__(
         if not np.issubdtype(time.dtype, np.datetime64):
             warnings.warn("Time data is not np.datetime64. Attempting to coerce.")
             try:
-                pd_time = pd.to_datetime(self["time"])
+                pd_time = _handle_time_column(pd.Series(self["time"]))
             except ValueError as e:
                 raise ValueError("Could not coerce time data to datetime64.") from e
             np_time = pd_time.to_numpy(dtype="datetime64[ns]")
             self.update(time=np_time)
         elif time.dtype != "datetime64[ns]":
             self.update(time=time.astype("datetime64[ns]"))
 
+        # In certain edge cases, "time" arrays will contain NaT enties (np.datetime64("NaT"))
+        # We don't want to support NaT entries
+        if np.any(np.isnan(self["time"])):
+            raise ValueError("NaT values are not allowed in the time field.")
```

Comment:
This is fairly strict .... I wonder if how necessary this is. We do this same [check in `Flight`](https://github.com/contrailcirrus/pycontrails/blob/main/pycontrails/core/flight.py#L243), but I think there's a more compelling reason for it there (for example, the `Flight` timestamps should be monotonically increasing). If you decide to keep this, you should remove the check in `Flight.__init__`.

Reply:
Oh good catch! I was copying this from the API approach - but you are right that we only need this for flights. I will remove it from here.

Reply:
e1143f209aa2fb7dade34ed59c0962f58203b65e